### DISCUSSION TITLE
How can i remove records of activities from my repository?

### DISCUSSION THREAD

**Discussion by tsp0proj** [community/community#134298] [Repositories] [answered]
### Select Topic Area

Question

### Body

There are a lot of unnecessary info about my commits, thats why i would like to delete these records. I tried to use this method:

```
git checkout --orphan new_branch (new branch)
git add . (root directory only)
git commit -m "message"
git branch -D main (delete main branch)
git branch -m main (rename current branch to 'main')
git push -f origin main (push and finish)
```

for deleting ununnecessary commits, but commits are saved anyway, in activities and also in "Actions" section. Please, help to get rid of it

**Comment by Buddhi19** [comment]
Well you can restore the version to a previous one,
```
git log --oneline
```
This will show your previous commit details with an Id, and to undo all the commits after a selected commit you can give the command,
```
git reset <commit ID> --hard
```
Hope this helps

**Reply by tsp0proj** [reply]
![image](https://github.com/user-attachments/assets/4dc11d01-0c9e-4a6d-b151-e79598fa6969)

Thank u, but there is only one commit, because i removed previous ones with git commands that is above in my question. So, how can i actually delete or change all my messages that i pinned to my commits from Activity and Action sections of github? 

![photo_2024-08-02_10-34-51](https://github.com/user-attachments/assets/1cd128bd-1052-40c2-82e2-301052788c50)
![photo_2024-08-02_10-34-45](https://github.com/user-attachments/assets/ca7ddf46-953b-414e-a588-afd00b7a7a75)

**Reply by Buddhi19** [reply, marked as answer]
Tried the second answer here?

https://stackoverflow.com/questions/13716658/how-to-delete-all-commit-history-in-github

**Reply by tsp0proj** [reply]
My project is already deployed and connected with netlify, if i will use this option, doesnt it lead to short circuit? As i understood, in this case, i will delete my old repo and create new one, and just will push my code again to the clear repo where history is clean, right?

**Reply by Buddhi19** [reply]
That should work!

**Reply by BhathiyaPrasad** [reply]
yeah, that will work but keep the copy of old one
